backend/app/moderation.py

```diff
@@ -0,0 +1,312 @@
+import base64
+import json
+import typing as T
+from datetime import datetime
+from enum import Enum
+
+import jwt
+from fastapi import APIRouter, Depends, FastAPI, HTTPException
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from fastapi_sqlalchemy import db as sqldb
+from pydantic import BaseModel, validator
+from sqlalchemy import func, or_
+
+from . import config, logins, models, worker
+from .db import get_json_key
+
+router = APIRouter(prefix="/moderation")
+
+
+class ModerationRequestType(str, Enum):
+    APPDATA = "appdata"
+
+
+def moderator_only(login=Depends(logins.login_state)):
+    if not login["state"].logged_in():
+        raise HTTPException(status_code=401, detail="not_logged_in")
+    if not login["user"].is_moderator:
+        raise HTTPException(status_code=403, detail="not_moderator")
+
+
+class ModerationAppItem(BaseModel):
+    appid: str
+    is_new_submission: bool
+    updated_at: datetime | None
+    request_types: list[ModerationRequestType]
+
+
+class ModerationAppsResponse(BaseModel):
+    apps: list[ModerationAppItem]
+
+
+class ModerationRequestResponse(BaseModel):
+    id: int
+    appid: str
+    created_at: int
+
+    build_id: int
+    job_id: int
+    is_outdated: bool
+
+    request_type: ModerationRequestType
+    request_data: T.Any | None
+    is_new_submission: bool
+
+    handled_by: str | None
+    handled_at: int | None
+    is_approved: bool | None
+    comment: str | None
+
+
+class ModerationApp(BaseModel):
+    requests: list[ModerationRequestResponse]
+
+
+@router.get("/apps", status_code=200, response_model_exclude_none=True)
+def get_moderation_apps(
+    new_submissions: bool | None = None, _moderator=Depends(moderator_only)
+) -> ModerationAppsResponse:
+    """Get a list of apps with unhandled moderation requests."""
+
+    is_new_submission = func.bool_or(models.ModerationRequest.is_new_submission).label(
+        "is_new_submission"
+    )
+    query = (
+        sqldb.session.query(
+            models.ModerationRequest.appid,
+            is_new_submission,
+            func.max(models.ModerationRequest.created_at).label("updated_at"),
+            func.array_agg(models.ModerationRequest.request_type.distinct()).label(
+                "request_types"
+            ),
+        )
+        .filter_by(handled_at=None, is_outdated=False)
+        .group_by(models.ModerationRequest.appid)
+    )
+
+    if new_submissions is not None:
+        query = query.having(is_new_submission == new_submissions)
+
+    return ModerationAppsResponse(
+        apps=[ModerationAppItem(**row._asdict()) for row in query]
+    )
+
+
+@router.get("/apps/{app_id}", status_code=200, response_model_exclude_none=True)
+def get_moderation_app(
```

Comment:
I think this should probably page, so that we don't run into problems, when an app has had some hundredth updates